### DISCUSSION TITLE
Adding New Food Commodities to Model

### DISCUSSION THREAD

**Discussion by junqi00** [magpiemodel/magpie#638] [Help] [answered]
Dear MAgPIE Community,

I encountered a challenge when using MAgPIE. I aim to incorporate additional processed food commodities into MAgPIE. I attempted to directly modify the model's input files in corresponding model folders, and modified part of the GAMS code to include these new commodities, but unfortunately, this led to errors when running the model.

Could you advise on the recommended procedure for introducing new food commodities into MAgPIE? Is manually editing specific files within the model the correct approach, or is there an alternative method I should consider?

I appreciate your time and look forward to your suggestions.

Thank you.

**Comment by tscheypidi** [comment, marked as answer]
Hi @junqi00,

manually editing the gms files should be fine, but if you want to change the input data, you should do it via patches (https://magpiemodel.github.io/tutorials/t12-changeinputs)

Best, Jan

**Reply by junqi00** [reply]
Dear Jan,

Thank you for your advice on using patches for MAgPIE. It worked perfectly, and I'm grateful for your help!

Best,
Junqi
